### DISCUSSION TITLE
netbox.dcim.interfaces.create issues

### DISCUSSION THREAD

**Discussion by twiclo** [netbox-community/pynetbox#559] [Q&A]
```
    return netbox.dcim.interfaces.create(
        {
            "device": "3215",
            "name": "Test" + str(rand),
            "type": "GigabitEthernet0/0/1"
            # "type": "Bridge"
        }
    )
```
That's the call I'm making to the create function and here's my response
`pynetbox.core.query.RequestError: The request failed with code 400 Bad Request: {'type': ['GigabitEthernet0/0/1 is not a valid choice.']}`
Where do I find the valid choice?
Caveat: This is pynetbox version 4.0.8. I understand that's old but I'm forced to work with legacy software until we can get it updated. I understand if help for something this old is hard to come by but any pointers will be appreciated

**Comment by K3Mi** [comment]
I'm also having a similar issue where I'm attempting to first check if a device is created and then check if interfaces are created for the device. during my testing, I'm getting the same error in my debug logs in Pynetbox version 7.2.0 
I did modify my code to something similar to the OP but still received the same error

ERROR: NetBox API request error: The request failed with code 400 Bad Request: {'type': ['This field is required.']}

```python
# here's my original code snippet

 # Get existing interfaces for the device in NetBox
    existing_interfaces = netbox.dcim.interfaces.filter(device_id=device.id)
    existing_interface_names = [interface.name for interface in existing_interfaces]

    # Add Interfaces to NetBox if they don't exist
    for interface in interfaces:
        if interface not in existing_interface_names:
            netbox.dcim.interfaces.create(device=device.id, name=interface)

```

**Reply by abhi1693** [reply]
Please refer to the schema for your netbox version. Here is an example link https://demo.netbox.dev/api/schema/swagger-ui/#/dcim/dcim_interfaces_create

**Reply by K3Mi** [reply]
Thanks, I did go back to the schema and modified my code as follows which worked. 

```python
 # Add Interfaces to NetBox if they don't exist
    for interface in interfaces:
        if interface not in existing_interface_names:
            netbox.dcim.interfaces.create(
                device=device.id, name=interface, type="virtual"
            )
```

**Comment by markkuleinio** [comment]
FWIW, with pynetbox 7.2.0 and NetBox 3.3:

```
>>> from pprint import pprint
>>> pprint(netbox.dcim.interfaces.choices()["type"])
[{'display_name': 'Virtual', 'value': 'virtual'},
 {'display_name': 'Bridge', 'value': 'bridge'},
 {'display_name': 'Link Aggregation Group (LAG)', 'value': 'lag'},
 {'display_name': '100BASE-TX (10/100ME)', 'value': '100base-tx'},
 {'display_name': '1000BASE-T (1GE)', 'value': '1000base-t'},
 {'display_name': '2.5GBASE-T (2.5GE)', 'value': '2.5gbase-t'},
 {'display_name': '5GBASE-T (5GE)', 'value': '5gbase-t'},
 {'display_name': '10GBASE-T (10GE)', 'value': '10gbase-t'},
 {'display_name': '10GBASE-CX4 (10GE)', 'value': '10gbase-cx4'},
 {'display_name': 'GBIC (1GE)', 'value': '1000base-x-gbic'},
...
```
